# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/tests/browser/e10s/browser_treeupdate_move.js
# Commit: 027ddef0767b
# Full Hash: 027ddef0767b43fa1f76ea7e56c1e04c64c27a88
# Author: James Teh <jteh@mozilla.com>
# Date: 2024-09-06 21:37:32
# Description:
#   Bug 1917132: When clearing a RemoteAccessible's parent, set its cached IndexInParent to -1. r=nlapre
#   
#   We set the cached IndexInParent to -1 in RemoveChild, among other places.
#   However, when a RemoteAccessible is moved, we clear its children array without calling RemoveChild (in DocAccessibleParent::ShutdownOrPrepareForMove).
#   We already call SetParent(nullptr) on the children to clear the parent, so set the cached IndexInParent to -1 in SetParent when the supplied new parent is null.
# ==============================================================================

diff -r 94c97b3688bc -r 027ddef0767b accessible/tests/browser/e10s/browser_treeupdate_move.js
--- a/accessible/tests/browser/e10s/browser_treeupdate_move.js	Fri Sep 06 09:39:03 2024 +0000
+++ b/accessible/tests/browser/e10s/browser_treeupdate_move.js	Fri Sep 06 09:54:46 2024 +0000
@@ -41,6 +41,7 @@
     await focused;
     testStates(textbox, STATE_FOCUSED, 0, 0, EXT_STATE_DEFUNCT);
 
+    let hidden = waitForEvent(EVENT_HIDE, textbox);
     let reordered = waitForEvent(EVENT_REORDER, docAcc);
     await invokeContentTask(browser, [], () => {
       // scrollable wasn't in the a11y tree, but this will force it to be created.
@@ -48,6 +49,8 @@
       content.document.getElementById("scrollable").style.overflow = "scroll";
       content.document.getElementById("heading").remove();
     });
+    await hidden;
+    ok(!para.nextSibling, "para nextSibling is null during move");
     await reordered;
     // Despite the move, ensure textbox is still alive and is focused.
     testStates(textbox, STATE_FOCUSED, 0, 0, EXT_STATE_DEFUNCT);
